clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -331,6 +331,11 @@ public static ByteBuffer wrapNullable(byte[] array) {
      */
     public static void sleep(long ms) {
         try {
+            if (ms <= 0) {
+                // No need to sleep
+                log.debug("Skipping sleep as asked to sleep for {} msec", ms);
```

Comment:
Won't this hide bugs?

Reply:
Yes. I have seen things done both ways i.e. behave reasonably in face of weird args vs throw exception. `Time.sleep` combines both approaches, throws for values `< 0` but skips if sleep `= 0` and goes to sleep for values `> 0`.

Its a judgement call, we can delegate to JDK and remove this check.

Reply:
Overflow is not uncommon when using things like Long.MaxValue, so assuming negative is not a bug seems dangerous.

Reply:
Ok, I see the point. Overflow can be caused by time arithmetic too. Reverted the change.